homeassistant/components/climate/maxcube.py

```diff
@@ -195,3 +198,25 @@ def map_mode_max_hass(mode):
             operation_mode = None
 
         return operation_mode
+
+class MaxCubeClimateValve(MaxCubeClimate):
+    """MAX! Cube Valve ClimateDevice."""
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS | SUPPORT_CURRENT_VALVEPOSITION
+
+    @property
+    def current_valveposition(self):
+        """Return the position of the valve."""
+        device = self._cubehandle.cube.device_by_rf(self._rf_address)
+        return self.map_valvepos_max_hass(device.valve_position)
+
+    @staticmethod
+    def map_valvepos_max_hass(position):
+        """Map Valveposition from MAX! to HASS."""
+        if position is None:
+            return 0.0
+
+        return position
```

Comment:
HI, can you please help mi how to implement in in prod? I changed climate/__init__.py climate/maxcube.py and still have no valve position. Thank you a log.

Reply:
Hello, did you put *climate/\_\_init\_\_.py* in your homeassistant folder?
*climate/maxcube.py* can be in either (homeassistant folder or *config/custom_components*).

Do you see the logs of the maxcube lib?
I have this in my log:
 ```
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] Cube (rf=162bec, firmware=01.13)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] WallThermostat (type=3, rf=088307, room=Keuken, name=Thermostaat, min=4.5, max=30.5, actual=20.0, target=17.0)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] Thermostat (type=2, rf=0AF04F, room=Keuken, name=Convector, mode=0, min=4.5, max=30.5, actual=None, target=17.0)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] WallThermostat (type=3, rf=0880A2, room=Woonkamer, name=Thermostaat, min=4.5, max=30.5, actual=18.0, target=17.0)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] Thermostat (type=2, rf=0AF063, room=Woonkamer, name=Convector achter, mode=0, min=4.5, max=30.5, actual=None, target=17.0)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] Thermostat (type=2, rf=0AF3EE, room=Woonkamer, name=Convector voor, mode=0, min=4.5, max=30.5, actual=None, target=17.0)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] WallThermostat (type=3, rf=08877B, room=Berging/Bureau, name=Thermostaat, min=4.5, max=30.5, actual=16.0, target=15.0)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] Thermostat (type=2, rf=0AEED6, room=Berging/Bureau, name=Radiator achter, mode=0, min=4.5, max=30.5, actual=None, target=15.0)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] Thermostat (type=2, rf=0AEE2F, room=Berging/Bureau, name=Radiator voor, mode=0, min=4.5, max=30.5, actual=None, target=15.0)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] WallThermostat (type=3, rf=088363, room=Hal, name=Thermostaat, min=4.5, max=30.5, actual=19.2, target=17.0)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] WallThermostat (type=3, rf=0882C0, room=WC, name=Thermostaat, min=4.5, max=30.5, actual=18.6, target=17.0)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] Thermostat (type=2, rf=0AF057, room=Hal, name=Convector, mode=0, min=4.5, max=30.5, actual=None, target=17.0)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] Thermostat (type=2, rf=0AEF69, room=Badkamer, name=Radiator, mode=0, min=4.5, max=30.5, actual=None, target=17.0)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] Thermostat (type=2, rf=0AEE56, room=WC, name=Radiator, mode=0, min=4.5, max=30.5, actual=None, target=17.0)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] Thermostat (type=2, rf=0AEF51, room=Slaapkamer 1, name=Convector, mode=0, min=4.5, max=30.5, actual=None, target=17.0)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] Thermostat (type=2, rf=0AEF28, room=Slaapkamer, name=Convector, mode=0, min=4.5, max=30.5, actual=None, target=17.0)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] Thermostat (type=2, rf=0AF3E2, room=Slaapkamer 2, name=Convector, mode=0, min=4.5, max=30.5, actual=None, target=17.0)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] WallThermostat (type=3, rf=08836F, room=Badkamer, name=Thermostaat, min=4.5, max=30.5, actual=17.5, target=17.0)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] WallThermostat (type=3, rf=08830C, room=Slaapkamer 1, name=Thermostaat, min=4.5, max=30.5, actual=18.2, target=17.0)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] WallThermostat (type=3, rf=088309, room=Slaapkamer 2, name=Thermostaat, min=4.5, max=30.5, actual=18.5, target=17.0)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] WallThermostat (type=3, rf=08830E, room=Slaapkamer, name=Thermostaat, min=4.5, max=30.5, actual=19.0, target=17.0)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] WindowShutter (type=4, rf=0732BC, room=Slaapkamer, name=Raam, init=8, open=False)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] WindowShutter (type=4, rf=073181, room=Badkamer, name=Raam, init=6, open=False)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] WindowShutter (type=4, rf=0731F5, room=Slaapkamer 2, name=Raam, init=9, open=False)
2018-03-19 02:04:43 INFO (SyncWorker_10) [maxcube.cube] WindowShutter (type=4, rf=0731FF, room=Slaapkamer 1, name=Raam, init=7, open=False)
 ```

Reply:
Thanks. I put it in config/custom_components and can see it in states of developement tools. Is it so that it is not implemented in GUI or History?  I got also errors when change few times temperature. Max!Cube connection failed. Do you got experience in this? I am currently migration from Pimatic and are new boy in HASS.

Reply:
Hello, I migrated from FHEM last weekend for my Max! stuff. I was using a CUL before. Only downside for now is the polling mechanism. It's not possible to trigger something by a Max! ecoswitch because of the polling. Beside that, it running very stable. Better than with CUL on FHEM (had sometimes problem with queue getting full without me doing anything).

For now, the valveposition is only an attribute. I would be nice to see it in the history tab. I made template sensors to achieve this. But maybe we can integrate this into the component/platform?

I use valveposition to switch on the central gasheater (sonoff with mqtt). It's switching when any valve is open. I used a binary-switch template to achieve this.

The Cube connection is very fragile. Avoid polling it to frequent or connection to it with multiple devices. I had a problem once but after a restart it was ok again.

Reply:
Hi, 
I would appreciate to integrate valve position in component so can see it in history tab. Can I ask you how to do configuration for template sensors as you mentioned?

My usage for valve position is also gasheater. Esp easy with relay controlled by mqtt. 

I have for testing purposes Virtual box with Centos and used NAT. Than I switched to network bridge in config and it looks stable. Lets see how this will work with my new Raspberry 3B+ just came.

Thank you a lot for exchanging you knowledge.

BR, Michal

Reply:
Hi Michal,

I created a template sensor for each thermostat.
```
sensor:
  - platform: yr
  - platform: template
    sensors:
      valve_badkamer_radiator:        
        friendly_name: "Kraan badkamer"
        unit_of_measurement: '%'
        value_template: "{{ states.climate.badkamer_radiator.attributes.current_valveposition }}"
        icon_template: >-
          {% if is_state_attr('climate.badkamer_radiator', 'current_valveposition', 0) %}
            mdi:checkbox-blank-circle
          {% else %} 
            mdi:checkbox-blank-circle-outline
          {% endif %} 
```

This sensor will show in the history tab.
I agree, it would be nicer to have it build into the climate platform.

You can make your own history groups like this:

```
history_graph:
  badkamer:
    name: Badkamer
    entities:
      - media_player.badkamer
      - light.badkamer
      - binary_sensor.badkamer_raam
      - climate.badkamer_thermostaat
      - sensor.valve_badkamer_radiator
    hours_to_show: 20
    refresh: 60
```
This looks like this:
![image](https://user-images.githubusercontent.com/3706036/37735537-a32b4c34-2d4e-11e8-942f-348e9cbb9589.png)
and if you have multiple thermostats in one room:
![image](https://user-images.githubusercontent.com/3706036/37735616-e0e7761a-2d4e-11e8-9c04-338c5200d54a.png)

I'm using this code to switch my gasheater:

```
binary_sensor:
  - platform: template
    sensors:
      kranen:
        friendly_name: "Kranen"
        device_class: opening
        value_template: >-
          {{ not (is_state_attr('climate.badkamer_radiator', 'current_valveposition', 0)
             and is_state_attr('climate.bergingbureau_radiator_achter', 'current_valveposition', 0)
             and ....) }}

- alias: Verwarming - aan
  trigger:
    platform: state
    entity_id: binary_sensor.kranen
    to: 'on'
  action:
    - service: mqtt.publish
      data_template:
        topic: 'cmnd/sonoff/POWER'
        payload: 'ON'

- alias: Verwarming - uit
  trigger:
    platform: state
    entity_id: binary_sensor.kranen
    to: 'off'
  action:
    - service: mqtt.publish
      data_template:
        topic: 'cmnd/sonoff/POWER'
        payload: 'OFF'
```

I use a minidesk PC with unraid with Hass running in docker.

BR, Maarten

Reply:
Hi Maarten,

Thank you for sharing this :)

BR
Michal